### PR TITLE
Task creation with popup

### PR DESCRIPTION
**Pull request by KeeyanGhoreshi**
Adds the create task button and attestation.  Needs a rebase.

### CODE REVIEW

**Comment by hossenlopp** [comment]
image

That button is so hard to find and in a random place.

File: src/components/Popup/TaskQuestionnaire.js [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+    },
+    "title": "Task",
+    "status": "draft",
+    "subjectType": ["Patient"],
+    "date": "2019-12-06",
+    "publisher": "Da Vinci DTR",
+    "extension": [
+      {
+        "url": "http://hl7.org/fhir/StructureDefinition/cqf-library",
+        "valueCanonical": "http://hl7.org/fhir/us/davinci-dtr/Library/HomeOxygenTherapy-prepopulation"
```
Comment (hossenlopp) [inline src/components/Popup/TaskQuestionnaire.js:19]: I'm sure these libraries do not need to be referenced for this Questionnaire.

File: src/components/Popup/TaskQuestionnaire.js [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+                "required": false,
+            },
+            {
+                "linkId": "2.8",
+                "text": "note",
+                "type": "text",
+                "required": false,
+                "initial": [
+                    {
+                        "valueString": "sample text"
```
Comment (hossenlopp) [inline src/components/Popup/TaskQuestionnaire.js:276]: Perhaps remove this sample text?

File: src/components/Popup/TaskQuestionnaire.js [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+                "text": "basedOn",
+                "type": "reference",
+                "required": false
+            },
+            {
+                "linkId": "1.2",
+                "text": "status",
+                "type": "choice",
+                "required": true,
+                "answerOption": [
```
Comment (hossenlopp) [inline src/components/Popup/TaskQuestionnaire.js:47]: The drop downs for these and other "choice" type questions here do not work in the UI. The only way to know a valid code to type in is to look at the source.

image

File: src/util/taskCreation.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+function createTask(info, smart) {
+
+
+    smart.patient.read().then((result) => {
+        info.resourceType = "Task";
+        info.status = info.status ? info.status : "draft";
+        info.intent = info.intent ? info.intent : "order";
+        info.for = info.for ? info.for : {reference:  `${result.resourceType}/${result.id}`};
```
[the remark this answers is not in the document]
Comment (KeeyanGhoreshi) [reply src/util/taskCreation.js:8]: this is just a check to ensure submitted tasks have required fields before submission.  Other data is already in the task and is optional, so it is not checked.

File: src/components/Popup/TaskQuestionnaire.js (hunk 3 above)
Comment (KeeyanGhoreshi) [reply src/components/Popup/TaskQuestionnaire.js:47]: strange, thought I had fixed that bug.

**Comment by plarocque4** [comment]
In the Task Form, should the window close when you click Submit?

**Comment by plarocque4** [comment]
Looks and works good.

**Pull request merged**